### ISSUE TITLE
enum names ignored

### ISSUE THREAD

**Issue by jmr** [cc65/cc65#1048]
The following code has [two errors](https://godbolt.org/z/XiY5iL): a redeclared enum type and an undeclared enum type:
```
enum e { x };
enum e { y };

int f() { return sizeof(enum undeclared); }
```

[No error or warning is currently issued.](https://godbolt.org/z/9dRMgA)

**Comment by acqn** [comment]
Enum types didn't get registered the same way as structs/unions are.

Fixed in my working copy.

**Comment by acqn** [comment]
Would we consider this resolved with PR #1121 merged?
